File: Languages/PHP/rand_avg.php
<?php
// Hello, World!
echo "Hello, World!\n";

function generateRandomNumbers() {
    $count = 1000;
    $outputDir = "../rand_avg output";
    $outputFile = $outputDir . "/random_numbers.txt";
    
    // Create output directory
    if (!file_exists($outputDir)) {
        mkdir($outputDir, 0755, true);
    }
    
    // Generate 1000 random numbers
    $randomNumbers = [];
    $sum = 0;
    
    for ($i = 0; $i < $count; $i++) {
        $num = rand(0, 999);
        $randomNumbers[] = $num;
        $sum += $num;
    }
    
    // Calculate mean
    $mean = $sum / $count;
    
    // Save to file
    $content = implode("\n", $randomNumbers);
    file_put_contents($outputFile, $content);
    
    echo "Generated 1000 random numbers\n";
    echo "Mean: " . number_format($mean, 2) . "\n";
    echo "Saved to: $outputFile\n";
}

generateRandomNumbers();
?>

File: Languages/Nim/rand_avg.nim
# Hello, World!
echo "Hello, World!"

import os, random, strutils

proc generateRandomNumbers() =
  let count = 1000
  let outputDir = "../rand_avg output"
  let outputFile = outputDir / "random_numbers.txt"
  
  # Create output directory
  createDir(outputDir)
  
  # Generate 1000 random numbers
  var randomNumbers: seq[int]
  var sum = 0
  
  for i in 0..<count:
    let num = rand(999)
    randomNumbers.add(num)
    sum += num
  
  # Calculate mean
  let mean = sum.float / count.float
  
  # Save to file
  let file = open(outputFile, fmWrite)
  for num in randomNumbers:
    file.writeLine($num)
  file.close()
  
  echo "Generated 1000 random numbers"
  echo "Mean: " & formatFloat(mean, ffDecimal, 2)
  echo "Saved to: " & outputFile

randomize()
generateRandomNumbers()
